File: AIZU/Advisor.java
import java.util.Scanner;

public class Advisor {
	public static int gcd(int x, int y) {
	        if (x % y == 0) return y;
       	 	else return gcd(y, x % y);
	}	
	public static void main(String argv[]) {
		Scanner in = new Scanner(System.in);
		int a = in.nextInt();
		int b = in.nextInt();
        	int d = gcd(a, b);
        	System.out.println(d);
	}
}

File: AIZU/advisor.cpp
#include <iostream>
using namespace::std;

int gcd(int x, int y) {
	if (x % y == 0) return y;
	else return gcd(y, x % y);
}

int main()
{
	int a, b;
	cin >> a;
	cin >> b;
	//cout<<"a " <<a<<" b "<<b<<endl;
	int d = gcd(a, b);	
	cout<<d<<endl;
}
